code/modules/atmospherics/machinery/portable/portable_atmospherics.dm

```diff
@@ -41,11 +43,21 @@
 	fire = 60
 	acid = 30
 
+/obj/machinery/portable_atmospherics/get_save_vars()
+	. = ..()
+	var/datum/gas_mixture/gasmix = air_contents
+	initial_gas_mix = gasmix.to_string()
+	. += NAMEOF(src, initial_gas_mix)
+	return .
+
 /obj/machinery/portable_atmospherics/Initialize(mapload)
 	. = ..()
-	air_contents = new
+	if(initial_gas_mix)
+		air_contents = SSair.parse_gas_string(initial_gas_mix)
+	else
+		air_contents = new
+		air_contents.temperature = T20C
```

Comment:
I do not like how much we're leaking map saving stuff into normal game logic

Reply:
I copied the code logic for how turfs initialize atmos and applied it to canisters. If you want me too, I can remove it since it's not that big of a deal. I did consider adding it to handheld air canisters but felt that would have been overkill because those aren't that important.

One thing I did forget to mention in our discord conversation is that extending map export/saving behavior is organically supporting VV edits for mappers which is a nice side benefit. Even if you gut all the `get_save_vars()` code, the code logic used makes sure VV edits are being applied correctly.